src/app/PCM/templates/tpl-sbe-row-add/sbe-add-element.js

```diff
@@ -17,7 +18,13 @@ const SBEAddElement = ({
 
   const component = askData?.question?.attribute?.dataType.component || ''
 
-  return equals(component)('dropdown') && !disabled ? (
+  if (component === '') {
+    debugOut.warn(`Component for ${attributeCode} was empty! Defaulting to a text component`)
+  }
+
+  const isComponent = compose(and(!disabled), equals(component))
+
+  return isComponent('dropdown') ? (
```

Comment:
You can use <Attribute /> component directly if it doesn't involve data interaction with the user for a component, which handles what kind of data type it is and renders the component based on that data type.

Reply:
The main reason I did this was to save a large amount of code duplication, I thought it looked a lot neater like this

Reply:
I think it is better to have the consistency through out the app and let the Attribute component handle all the non data interactive components. It is more declarative in the sense that, it is supposed to do that through out the app. But if there are duplications of code, we can create an adapter component function around it to handle that, which is a nice idea. 
Just curious, what kind of code duplication it is.

Reply:
Oh I more meant I found it frustrating to write 'If not disabled, and this component" for every line. I found `isComponent` was a quick and elegant way to reduce how much I had to write on each line